src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/MsQuicNativeMethods.cs

```diff
@@ -272,6 +275,17 @@ internal struct CredentialConfigCertificateCertificateFileProtected
             internal string PrivateKeyPassword;
         }
 
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct CredentialConfigCertificatePkcs12
+        {
+            internal IntPtr Asn1Blob;
+
+            internal int Asn1BlobLength;
```

Comment:
shouldn't this be a `uint`?

Reply:
That is generally the default for .NET -> use int unless there is reason no to. I personally don't care either way.

Reply:
We mainly use uint when necessary to match native headers.

Reply:
array.Length is `int` so as pretty much all `Length` properties in c#. We will have to cast and the outcome would be same. 
This is pretty common that underlying `C` uint is exposed as `int` to make c# code more natural. For example 

https://github.com/dotnet/runtime/blob/0423ac97a6dcb54a9f096f848cd20346217f0c6a/src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs#L54-L55

and 

https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/Native/Unix/System.Security.Cryptography.Native/pal_ssl.h#L366

cc: @stephentoub for any thoughts.  I'm fine if we embark on the casting route as this is all internal.

Reply:
I'm okay with us using `int` for buffer lengths. I don't know why `uint` is being used so excessively in the code right now, it doesn't really add value. Whatever we do we should be consistent.

Reply:
One of our goals with our interop files is that they should be as close to the native API signature as possible,  e.g. as if we were able to just include the .h file and target its signature directly.  So we've used uint (or at least intended to) where the underlying signature is for an unsigned int,  even if that complicates the interop consumption slightly.

Reply:
updated to `uint`